src/main.jsx

```diff
@@ -0,0 +1,1102 @@
+
+  const [microscopeControl, setMicroscopeControl] = useState(null);
+  const [snapshotImage, setSnapshotImage] = useState(null);
+
+  const [xPosition, setXPosition] = useState(0);
+  const [yPosition, setYPosition] = useState(0);
+  const [zPosition, setZPosition] = useState(0);
+  const [BrightFieldIntensity, setBrightFieldIntensity] = useState(50);
+  const [BrightFieldCameraExposure, setBrightFieldCameraExposure] = useState(100);
+  const [Fluorescence405Intensity, setFluorescence405Intensity] = useState(50);
+  const [Fluorescence405CameraExposure, setFluorescence405CameraExposure] = useState(100);
+  const [Fluorescence488Intensity, setFluorescence488Intensity] = useState(50);
+  const [Fluorescence488CameraExposure, setFluorescence488CameraExposure] = useState(100);
+  const [Fluorescence561Intensity, setFluorescence561Intensity] = useState(50);
+  const [Fluorescence561CameraExposure, setFluorescence561CameraExposure] = useState(100);
+  const [Fluorescence638Intensity, setFluorescence638Intensity] = useState(50);
+  const [Fluorescence638CameraExposure, setFluorescence638CameraExposure] = useState(100);
+  const [Fluorescence730Intensity, setFluorescence730Intensity] = useState(50);
+  const [Fluorescence730CameraExposure, setFluorescence730CameraExposure] = useState(100);
+  const [illuminationIntensity, setIlluminationIntensity] = useState(50);
+  const [illuminationChannel, setIlluminationChannel] = useState("0");
+  const [cameraExposure, setCameraExposure] = useState(100);
+  const [xMove, setXMove] = useState(1);
+  const [yMove, setYMove] = useState(1);
+  const [zMove, setZMove] = useState(0.1);
+  const [isLightOn, setIsLightOn] = useState(false);
+  const [isPlateScanRunning, setIsPlateScanRunning] = useState(false);
+  const [log, setLog] = useState('');
+  const [chatbotUrl, setChatbotUrl] = useState(null);
+  const [svc, setSvc] = useState(null);  // Correctly initialize the similarity search service state
+  const [numSimilarResults, setNumSimilarResults] = useState(5);
+  const [searchResults, setSearchResults] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isPenActive, setIsPenActive] = useState(false); // State to track if the pen tool is active
+  const [isFirstClick, setIsFirstClick] = useState(true); // Track if it's the first click for segmentation
+  const [segmentService, setSegmentService] = useState(null);
+  const [selectedModel, setSelectedModel] = useState('vit_b_lm'); // Default model
+  const [similarityService, setSimilarityService] = useState(null);
+  const appendLog = (message) => {
+      setLog(prevLog => prevLog + message + '\n');
+  };
+
+  useEffect(() => {
+    const initializeWebRPC = async () => {
+        try {
+            appendLog('Connecting to server...');
+            const server = await hyphaWebsocketClient.connectToServer({"server_url": "https://hypha.aicell.io", "webrtc": true})                                  
```

Comment:
@oeway Is this correct?

Reply:
The issue is that you connect to the server as anonymous user (from user's own workspace), and when you call `getSerivce("interactive-segmentation")`, the relative service id "interactive-segmentation" assumes you have that service registered under the anonymous user's own workspace. This is why it gives you service not found. You should use the full service id in the format of "{workspace}/interactive-segmentation", you need to get the actual workspace id where you registered `interactive-segmentation` to replace `{workspace}`.

Another note, please implement a login feature following:
https://docs.amun.ai/#/getting-started?id=user-login-and-token-based-authentication
We don't want anyone has access to the service eventually.